notebooks/04-esmvaltool-singularity/04-esmvaltool-singularity.ipynb

```diff
@@ -0,0 +1,285 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "id": "fc84c549-1e95-4aec-b9fc-c56a16024600",
+   "metadata": {},
+   "source": [
+    "# Demo: Run an esmvaltool recipe using containers"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "e0e93897-8e1a-4029-a9ef-9996f8fd56cb",
+   "metadata": {},
+   "source": [
+    "In this example, we demonstrate how to execute an ESMValtool recipe using containers on Spiders infrastructure."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "c9bb6d31",
+   "metadata": {},
+   "source": [
+    "## Prerequest: Starting a Jupyter Server on Spider.\n",
+    "\n",
+    "A Jupyter Server should be started on Spider as the environment to execute this notebook. You can follow [this instruction](https://github.com/RS-DAT/JupyterDaskOnSLURM) to start a Jupyter server on Spider.\n",
+    "\n",
+    "After successfully setting up the Jupyter server, please copy this notebook to the Spider file system. Then open it from the browser on your local PC (as specified in the above instructon). Effectively, this notebook run as a Slurm job on Spider."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "bf44d734",
+   "metadata": {},
+   "source": [
+    "## Step 1: build a Singularity container\n",
+    "\n",
+    "ESMValTool is provided in the form of Docker containers. However on Spiders, [it is suggested to use Singularity](https://spiderdocs.readthedocs.io/en/latest/Pages/software_on_spider.html?highlight=singularity#singularity-containers) as the container technology. Therefore we will convert the existing Docker container to a Singularity container. Please run this command on Spiders:\n",
+    "\n",
+    "```sh\n",
+    "# Step1: build sif image (this should be done once)\n",
+    "singularity build esmvaltool_stable.sif docker://esmvalgroup/esmvaltool:stable\n",
+    "```\n",
+    "\n",
+    "This will build a Singularity Image File (.sif) named `esmvaltool_stable.sif` on the Spiders file system, from DockerHub. Note that this may take ~20 minutes."
```

Comment:
Spiders -> Spider

We should maybe also stress that the conversion of Docker to Singularity runs smoothly for most/all containers. Are we aware of known exceptions caveats?